packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol

```diff
@@ -803,9 +839,46 @@ contract OPContractsManagerV2 is ISemver, OPContractsManagerUtilsCaller {
             abi.encodeCall(IDelayedWETH.initialize, (_cts.systemConfig))
         );
 
+        // Update the ETHLockbox, DisputeGameFactory, and AnchorStateRegistry.
+        // If this is an interop migration upgrade, these contracts may have a different
+        // ProxyAdmin, but regardless all chains migrating to an interop set MUST have the
+        // same ProxyAdmin owner (which is delegate calling this contract).
+        // Therefore for the next set of upgrades, we use either:
+        //   1. The current chain's proxy admin for initial deployments
+        //   2. The ETHLockbox's proxyAdmin() getter for upgrades and interop migrations (if lockbox exists)
+        //   3. The current chain's proxy admin for upgrades when lockbox doesn't exist
+        // It would be ideal to simply use the proxyAdmin() getter in all cases, however for initial deployments
```

Comment:
I unfortunately cannot think of any nicer way to handle this other than ensuring that all proxies get initialized to some basic contract that has no functionality except exposing `proxyAdmin` when they first get created